File: final_dataset_A/generated_code/chatgpt/java/OrderStatusUpdateEndpointDetailed-chatgpt.java
package com.example.demo.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.time.LocalDateTime;
import java.util.LinkedHashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/orders")
public class OrderController {

    @PutMapping("/{orderId}/status")
    public ResponseEntity<Map<String, Object>> updateOrderStatus(
            @PathVariable Long orderId,
            @RequestParam String status) {

        // Simulate updating the order
        Map<String, Object> order = new LinkedHashMap<>();
        order.put("orderId", orderId);
        order.put("status", status);
        order.put("updatedAt", LocalDateTime.now());

        // Confirmation response
        Map<String, Object> response = new LinkedHashMap<>();
        response.put("success", true);
        response.put("message", "Order status updated successfully.");
        response.put("order", order);

        return ResponseEntity.ok(response);
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/order_status_update_endpoint_detailed-chatgpt.py
from flask import Flask, request, jsonify
from datetime import datetime

app = Flask(__name__)

@app.route("/orders/<int:order_id>/status", methods=["PUT"])
def update_order_status(order_id):
    # Get the new status
    data = request.get_json()
    status = data.get("status")

    # Simulate updating the order
    order = {
        "orderId": order_id,
        "status": status,
        "updatedAt": datetime.now().isoformat()
    }

    # Confirmation response
    response = {
        "success": True,
        "message": "Order status updated successfully.",
        "order": order
    }

    return jsonify(response), 200

if __name__ == "__main__":
    app.run(debug=True)